script-library/dotnet-debian.sh

```diff
@@ -0,0 +1,245 @@
+# NOTE: icu-devtools is a small package with similar dependecies to dotnet. 
+#       It will install the appropriate dependencies based on the OS:
+#         - libgcc-s1 OR libgcc1 depending on OS
+#         - the latest libicuXX depending on OS (eg libicu57 for stretch)
+#         - also installs libc6 and libstdc++6 which are required by dotnet 
+check_packages curl ca-certificates tar icu-devtools libgssapi-krb5-2 libssl1.1 zlib1g
+
+# Set architecture variable to current user's architecture (x64 or ARM64).
+architecture="$(uname -m)"
+case $architecture in
+    x86_64) architecture="x64";;
+    aarch64 | armv8*) architecture="arm64";;
+    *) err "Architecture ${architecture} unsupported"; exit 1 ;;
+esac
+
+# Get the latest version for dotnet Runtime for use in generating the checksum URL 
+runtime_full_version=$(get_latest_version "Runtime")
+
+# Determine if the user wants to download dotnet Runtime only, or dotnet SDK & Runtime
+# and set the appropriate variables.
+if [ "${DOTNET_RUNTIME_ONLY}" = "true" ]; then
+    DOTNET_SDK_OR_RUNTIME="Runtime"
+    DOTNET_FULL_VERSION="${runtime_full_version}"
+elif [ "${DOTNET_RUNTIME_ONLY}" = "false" ]; then
+    DOTNET_SDK_OR_RUNTIME="Sdk"
+    DOTNET_FULL_VERSION=$(get_latest_version "Sdk")
+else
+    err "Expected true for installing dotnet Runtime only or false for installing SDK and Runtime. Received ${DOTNET_RUNTIME_ONLY}."
+    exit 1
+fi
+
+# Install the CLI
+echo "(*) Installing dotnet CLI..."
+
+# Setup the access group and add the user to it.
+umask 0002
+if ! cat /etc/group | grep -e "^${ACCESS_GROUP}:" > /dev/null 2>&1; then
+    groupadd -r "${ACCESS_GROUP}"
+fi
+usermod -a -G "${ACCESS_GROUP}" "${USERNAME}"
+
+# Get download link for dotnet binaries from azureedge.
+binary_download_link=$(get_download_link "${DOTNET_SDK_OR_RUNTIME}" "${DOTNET_FULL_VERSION}" "${architecture}")
+echo "DOWNLOAD LINK: ${binary_download_link}"
+
+
+# Download the dotnet binaries.
+binary_file_name=$(get_filename_from_url "${binary_download_link}")
+echo "BINARY FILE NAME: ${binary_file_name}"
+echo "DOWNLOADING BINARIES..."
+TMP_DIR="/tmp/dotnetinstall"
+mkdir -p "${TMP_DIR}"
+curl -sSL "${binary_download_link}" -o "${TMP_DIR}/${binary_file_name}"
+
+# Get checksum from dotnet CLI blob storage using the runtime version and
+# run validation (sha512sum) of checksum against the expected list of checksums.
+echo "FETCH AND VERIFY CHECKSUM"
+curl -sSL "https://dotnetcli.blob.core.windows.net/dotnet/checksums/${runtime_full_version}-sha.txt" -o "${TMP_DIR}/checksums.txt"
+grep "${binary_file_name}" "${TMP_DIR}/checksums.txt" | sed 's/\r$//' > "${TMP_DIR}/${binary_file_name}.sha512"
+cd "${TMP_DIR}/"
+sha512sum -c "${binary_file_name}.sha512"
+
+# Extract binaries and add to path.
+mkdir -p "${TARGET_INSTALL_PATH}"
+echo "Extract Binary to ${TARGET_INSTALL_PATH}"
+tar -xzf "${TMP_DIR}/${binary_file_name}" -C "${TARGET_INSTALL_PATH}" --strip-components=1
+
```

Comment:
suggestion: Remove all downloaded assets (tar.gz and checksum) after consumed/used.

Reply:
Agreed!  It is already here

https://github.com/microsoft/vscode-dev-containers/pull/1210/files#diff-323a2d3bd21416cfbcfa258c8a9ac0c569a8a99bfa2b09687aac3d3038634814R59-R69